packages/react/src/types/index.ts

```diff
@@ -47,6 +47,25 @@ declare module 'ethers/lib/utils.js' {
   ): string
 }
 
+/**
+ * Makes {@link TKeys} optional in {@link TType} while preserving type inference.
+ */
+// s/o trpc (https://github.com/trpc/trpc/blob/main/packages/server/src/types.ts#L6)
+export type InferOptional<TType, TKeys extends keyof TType> = Partial<
```

Comment:
nit: `PartialBy`? This feels like a variant of `Partial`, except, we are targetting provided `TKeys`.